core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1779,101 +2062,167 @@ class ReplicaManager(val config: KafkaConfig,
     partitionsToMakeLeaders
   }
 
-  private def makeFollowers(builder: MetadataImageBuilder,
+  private def makeFollowers(prevPartitionsAlreadyExisting: Set[MetadataPartition],
+                            currentBrokers: MetadataBrokers,
                             partitionStates: Map[Partition, MetadataPartition],
                             highWatermarkCheckpoints: OffsetCheckpoints,
-                            metadataOffset: Long): Set[Partition] = {
+                            defaultMetadataOffset: Long,
+                            metadataOffsets: Map[Partition, Long] = Map.empty): Set[Partition] = {
```

Comment:
It looks like we are only using this collection as well as `defaultMetadataOffset` for the purpose of logging. Is that right? We have one log message already in `makeDeferred` which indicates the offset when the update was received, so that makes me feel that this is probably overkill. Perhaps rather than indicating the exact offset in which the change was present, we can treat this more as the latest state at the time that the last offset was applied. Otherwise, having two collections keyed by `Partition` is suboptimal.

An alternative is that we could try to keep track of the offset in `MetadataPartition`.